app/api/helpers/storage.py

```diff
@@ -179,7 +183,12 @@ def upload_local(uploaded_file, key, **kwargs):
 
 def create_url(request_url, file_relative_path):
     """Generates the URL of an uploaded file."""
-    url = urlparse(request_url)
+
+    if env('APP_CONFIG') == 'config.ProductionConfig' or \
+            os.getenv('APP_CONFIG') == 'config.ProductionConfig':
+        url = urlparse('http://api.eventyay.com')
```

Comment:
As suspected, your previous methods(all) were wrong ways to produce the URLs

This too is wrong. What about the server hosted on heroku? Or self hosted ones. This is called "HARDCODING" which is the next bad thing about a server after deliberately throwing an exception

Reply:
I find two solutions of this, either using HARDCODING way or SERVER_NAME env.
You are not favour of using SERVER_NAME env too.
Please suggest a way which can fully solve the issue without these two ways.

> What about the server hosted on heroku? 

Heroku using DevelopmentConfig

![screenshot from 2018-08-10 20-33-11](https://user-images.githubusercontent.com/22936570/43965359-a6461036-9cdc-11e8-8e14-16546426f298.png)